lib/ansible/executor/process/worker.py

```diff
@@ -99,15 +119,16 @@ def start(self):
         make sure it is closed in the parent when start() completes.
         """
 
-        self._save_stdin()
         # FUTURE: this lock can be removed once a more generalized pre-fork thread pause is in place
         with display._lock:
-            try:
-                return super(WorkerProcess, self).start()
-            finally:
-                self._new_stdin.close()
-
-    def _hard_exit(self, e):
+            super(WorkerProcess, self).start()
+        # Since setsid is called later, if the worker is termed
+        # it won't term the new process group
+        # register a handler to propagate the signal
+        signal.signal(signal.SIGTERM, self._term)
+        signal.signal(signal.SIGINT, self._term)
```

Comment:
Is handling `SIGINT` needed? Since `tty` is detached after `setsid()` it shouldn't be possible for the user to do `Ctrl+C`, no? Or is this just being defensive, or?

Reply:
The `start` method is run in the main process and is registering a `SIGINT` handler in the main process to pass through the signal to the child pid. `self.pid` above is the child pid, accessible in the parent.

This is so that if `Ctrl+C` is done on the CLI of the main process that it is sent to the forks and any subprocesses of the fork.